### PR TITLE
Add NFData instances to Closure

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

**Comment by facundominguez** [comment]
What is this needed for?

Also, the `NFData` instance is not forcing the arguments of the `Closure` data constructor.

File: src/Control/Distributed/Static.hs [hunk 1]
```diff
@@ -348,6 +349,8 @@ instance Typeable a => Binary (Closure a) where
   put (Closure static env) = put static >> put env
   get = Closure <$> get <*> get
 
+instance NFData (Closure a) where rnf x = x `seq` ()
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]: I was expecting something like:
```
instance NFData (Closure a) where rnf (Closure st b) = rnf st `seq` rnf b
```

**Comment by qnikst** [comment]
This instance is required in order to write correct `NFData` instances in order packages, namely `distributed-process-extra`.

**Comment by qnikst** [comment]
@facundominguez can you recheck this Pull request?

**Comment by qnikst** [comment]
According to travis it doesn't work for 7.4 because ByteString.Lazy do not have `NFData` instance. The only way to fix it is to add a condition on the ByteString version and manyally force all chunks by taking length, or forbid old versions.

File: src/Control/Distributed/Static.hs [hunk 2]
```diff
@@ -257,10 +258,17 @@ data StaticLabel =
   | StaticApply StaticLabel StaticLabel
   deriving (Eq, Ord, Typeable, Show)
 
+instance NFData StaticLabel where
+  rnf (StaticLabel s) = rnf s `seq` ()
+  rnf (StaticApply a b) = rnf a `seq` rnf b `seq` ()
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]:
```Haskell
- rnf (StaticApply a b) = rnf a `seq` rnf b `seq` ()
+ rnf (StaticApply a b) = rnf a `seq` rnf b
```

File: src/Control/Distributed/Static.hs [hunk 3]
```diff
@@ -257,10 +258,17 @@ data StaticLabel =
...
+
 -- | A static value. Static is opaque; see 'staticLabel' and 'staticApply'.
 newtype Static a = Static StaticLabel
   deriving (Eq, Ord, Typeable, Show)
 
+instance NFData (Static a) where
+  rnf (Static s) = rnf s `seq` ()
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]:
```Haskell
+  rnf (Static s) = rnf s `seq` ()
-  rnf (Static s) = rnf s
```

File: src/Control/Distributed/Static.hs [hunk 4]
```diff
@@ -348,6 +356,8 @@ instance Typeable a => Binary (Closure a) where
   put (Closure static env) = put static >> put env
   get = Closure <$> get <*> get
 
+instance NFData (Closure a) where rnf (Closure f b) = rnf f `seq` rnf b `seq` ()
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]:
```Haskell
+ instance NFData (Closure a) where rnf (Closure f b) = rnf f `seq` rnf b `seq` ()
- instance NFData (Closure a) where rnf (Closure f b) = rnf f `seq` rnf b
```

**Comment by facundominguez** [comment]
> add a condition on the ByteString version and manyally force all chunks by taking length

I'd prefer this.

**Comment by qnikst** [comment]
@facundominguez comments were addressed and now PR pass tests.

**Comment by facundominguez** [comment]
Merged as feef72e.

**Pull request closed without merging**